src/main/resources/hudson/tasks/junit/History/index.jelly

```diff
@@ -20,88 +20,144 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
       <c:chart-setup id="test-history"/>
-      <script>var view =
-        <st:bind value="${it}"/>
+      <script>
+      var view = <st:bind value="${it}"/>
+      var start = ${start}
+      var end = ${end}
+      var trendChartJson = ${trendChartJson}
+      var trendChartJsonStr = ${trendChartJsonStr}
+      var rootUrl = "${app.rootUrl}"
       </script>
       <script type="text/javascript" src="${resURL}/plugin/junit/history/history.js"/>
     </st:once>
 
     <st:include from="${it.testObject}" it="${it.testObject}" page="sidepanel.jelly"/>
     <l:main-panel>
-      <j:choose>
-        <j:when test="${it.historyAvailable()}">
-          <bs:card title="${%History}" fontAwesomeIcon="chart-line">
-            <div id="trend-carousel" class="carousel slide carousel-dark" data-bs-interval="false">
-              <div class="carousel-inner">
-                <div class="carousel-item">
-                  <div id="test-duration-trend-chart" class="graph-cursor-pointer card-chart-carousel"/>
-                </div>
-                <div class="carousel-item active">
-                  <div id="test-result-trend-chart" class="graph-cursor-pointer card-chart-carousel"/>
-                </div>
-              </div>
-              <button class="carousel-control-prev" type="button" data-bs-target="#trend-carousel" data-bs-slide="prev">
-                <span class="carousel-control-prev-icon" aria-hidden="true"/>
-                <span class="visually-hidden">Previous</span>
-              </button>
-              <button class="carousel-control-next" type="button" data-bs-target="#trend-carousel" data-bs-slide="next">
-                <span class="carousel-control-next-icon" aria-hidden="true"/>
-                <span class="visually-hidden">Next</span>
-              </button>
-            </div>
-          </bs:card>
+      <div style="margin-bottom: 0.5em; margin-top: -1.75em; height: 1.5em;">
+        <j:if test="${it.testObject.run.parent.builds.size() > end}">
+          <a href="${app.rootUrl}${it.testObject.url}/history?start=${end+1}&amp;count=${count}"
+              style="margin:1px; margin-right: 10px;">${%Older}
+          </a>
+        </j:if>
 
-        </j:when>
-        <j:otherwise>
-          ${%More than 1 builds are needed for the chart.}
-        </j:otherwise>
-      </j:choose>
+        <j:if test="${start > 0}">
+          <a href="${app.rootUrl}${it.testObject.url}/history${(start-count)>0?'?start='+(start-count)+'&amp;':'?'}count=${count}"
+              style="margin:1px">${%Newer}
+          </a>
+        </j:if>
 
-      <j:set var="historySummary" value="${it.retrieveHistorySummary(start)}"/>
+        <div style="float: right; margin-left: auto; margin-bottom: 7px;">
+          Builds (only builds with test results are shown): 
+          <a href="${app.rootUrl}${it.testObject.url}/history?start=${start}&amp;count=100" style="margin:1px; margin-left: 10px;${count==100?'font-weight: bold;':''}">100</a>
```

Comment:
This should be more dynamic, my test job only has 22 results why do I see all this?

<img width="590" alt="image" src="https://github.com/jenkinsci/junit-plugin/assets/21194782/d42eb69c-ce53-4110-ba22-1f334439be8e">

Reply:
Since only builds with test results are shown, it would cause us to load those builds. We need to avoid that. I am not a fan of checking counts in Jenkins, because that does round trips to the build.xml of each build in some cases.

Reply:
Right I can't see a way to get the information, you are supposed to get the last one and iterate backwards but performance wise this wouldn't work.

can we drop the `(only builds with tests results are shown)` (maybe move it to a `tooltip` attribute)

and then yes move this to a dropdown.

Could you clarify exactly what this is for? is it show last 2000 builds (that have a test result) ?

Reply:
I will see if I can move it to a dropdown.

As far as I remember, it takes 2000 builds and only shows those that have test results.

Reply:
Hmm, I looked at select and overflowButton in Design Library and it seems that they are intended to interact with the Java side, but there is no such thing here directly. That would be overkill. 

Any tips?

Reply:
I did some improvements; I think a dropdown could be optional now...

Reply:
I think you could add the html elements with the styling from the design library and then add an event listener in JavaScript for change and then update the page's href from Javascript

Reply:
I am not a fan of this, because:

![image](https://github.com/user-attachments/assets/ccc8049f-2100-4f86-93c5-015306394c40)

Somehow it does not use the dark styles?

Reply:
can't you use a normal `<select>` element instead of the overflow?

Reply:
Are you referring to the plain HTML <select> or this: https://weekly.ci.jenkins.io/design-library/Select/ ?

It is slightly confusing, perhaps the Design Library should include an example with a plain HTML element, because now I don't know which styles to use.

Reply:
So the rendered html from https://weekly.ci.jenkins.io/design-library/Select/

this is quite a niche case as a lot of the design library is for forms whereas this is for a client side control